File: AddressBook/react-client/src/dtos/ContactPhone.ts
import Common from "../utility/Common";
import { CountryRes } from "./Country";
import { PagedReq } from "./PagedReq";
import { ContactRes } from "./Contact";
import { PhoneLabelRes } from "./PhoneLabel";

export interface ContactPhoneRes {
  contactPhoneId?: string;
  phone?: string;
  countryId?: string;
  country?: CountryRes;
  contactId?: string;
  contact?: ContactRes;
  phoneLabelId?: string;
  phoneLabel?: PhoneLabelRes;
}

export class ContactPhoneReqEdit {
  phone?: string = "";
  countryId?: string = "";
  contactId?: string = "";
  phoneLabelId?: string = "";
  constructor(contactId?: string) {
    this.contactId = contactId;
  }
}

export class ContactPhoneReqSearch extends PagedReq {
  contactId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy = "",
      searchText = "",
    }: PagedReq,
    {contactId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.contactId = contactId;
  }
}

File: AddressBook/AddressBook/Dtos/ContactPhone.cs
using AddressBook.Entities;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using AddressBook.Common.Paging;

namespace AddressBook.Dtos
{
    public class ContactPhoneRes
    {
        public int ContactPhoneId { get; set; }
        public string? Phone { get; set; }

        public int? CountryId { get; set; }
        public Country? Country { get; set; }

        public int ContactId { get; set; }
        public ContactRes? Contact { get; set; }

        public int? PhoneLabelId { get; set; }
        public PhoneLabelRes? PhoneLabel { get; set; }
    }

    public class ContactPhoneReqEdit
    {
        [Required]
        public string? Phone { get; set; }
        // Foreign keys
        public int? CountryId { get; set; }
        [Required]
        public int? ContactId { get; set; }
        public int? PhoneLabelId { get; set; }
    }

    public class ContactPhoneReqSearch : PagedReq
    {
        [Required]
        public int? ContactId { get; set; }
    }
}
